crates/cdk-common/src/mint.rs

```diff
@@ -746,3 +746,205 @@ mod offer_serde {
         })?))
     }
 }
+
+/// Batch Mint Request [NUT-XX]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(feature = "swagger", derive(utoipa::ToSchema))]
+pub struct BatchMintRequest {
+    /// Quote IDs
+    pub quote: Vec<String>,
+    /// Blinded messages
+    pub outputs: Vec<cashu::nuts::nut00::BlindedMessage>,
+    /// Signatures for NUT-20 locked quotes (optional)
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub signature: Option<Vec<Option<String>>>,
+}
+
+impl BatchMintRequest {
+    /// Total amount of outputs
+    pub fn total_amount(&self) -> Result<Amount, cashu::nuts::nut04::Error> {
+        Amount::try_sum(self.outputs.iter().map(|msg| msg.amount))
+            .map_err(|_| cashu::nuts::nut04::Error::AmountOverflow)
+    }
+}
+
+/// Batch Quote Status Request [NUT-XX]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(feature = "swagger", derive(utoipa::ToSchema))]
+pub struct BatchQuoteStatusRequest {
+    /// Quote IDs
+    pub quote: Vec<String>,
+}
+
+/// Bolt12 batch status payload extends the standard Bolt12 quote with a derived state.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(feature = "swagger", derive(utoipa::ToSchema))]
+#[serde(bound = "Q: Serialize + for<'a> Deserialize<'a>")]
+pub struct MintQuoteBolt12BatchStatusResponse<Q> {
+    /// Underlying Bolt12 quote payload
+    #[serde(flatten)]
+    pub quote: MintQuoteBolt12Response<Q>,
+    /// Derived quote state (UNPAID, PAID, ISSUED)
+    pub state: MintQuoteState,
+}
+
+impl<Q> MintQuoteBolt12BatchStatusResponse<Q> {
+    fn from_quote(quote: MintQuoteBolt12Response<Q>) -> Self {
+        let state = derive_quote_state(quote.amount_paid, quote.amount_issued);
+        Self { quote, state }
+    }
+
+    /// Current quote state
+    pub fn state(&self) -> MintQuoteState {
+        self.state
+    }
+}
+
+impl<Q> From<MintQuoteBolt12Response<Q>> for MintQuoteBolt12BatchStatusResponse<Q> {
+    fn from(value: MintQuoteBolt12Response<Q>) -> Self {
+        Self::from_quote(value)
+    }
+}
```

Comment:
Requests and responses defined in the nuts should be in the cashu crate in a mod for the nut, nutxx for now.